contrib/golang/.gitignore

```diff
@@ -0,0 +1 @@
+go.sum
```

Comment:
this doesnt seem correct - there are other `go.sum` files beneath this directory

also - i think if we are going to ignore something it should be in the main `.gitignore` at the root of the rpo

Reply:
correction - there are only `go.mod` files beneath this directory - but i think the point about not adding another .gitignore stands

Reply:
@phlax 
What about using this?
```
/contrib/golang/**/test_data/go.sum
/contrib/golang/**/test_data/*/go.sum
```

Reply:
sgtm

Reply:
/wait

Reply:
@phlax 
Solved in new commit